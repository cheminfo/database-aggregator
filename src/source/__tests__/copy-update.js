'use strict';

const mongoSetup = require('../../../test/mongoSetup');
const { getCollection, clean } = require('../../../test/util');
const copy = require('../copy');

beforeAll(mongoSetup.connect);
afterAll(mongoSetup.disconnect);

const driver = {
  getData(config, callback) {
    return callback([config.entry]);
  },
  getIds() {
    throw new Error('unimplemented');
  }
};

function getEntryCopy(collection, entry) {
  return copy({
    driver,
    collection,
    entry
  });
}

describe('source copy updates', () => {
  it('should not allow to change the commonID', async () => {
    await mongoSetup.insertData({
      // eslint-disable-next-line camelcase
      source_test1: [
        {
          id: 'test1',
          commonID: 'test1',
          sequentialID: 0,
          data: {},
          date: new Date(0)
        }
      ]
    });

    await expect(
      getEntryCopy('test1', {
        id: 'test1',
        commonID: 'test1-updated',
        data: {},
        modificationDate: new Date(1)
      })
    ).rejects.toThrow(/^commonID may not be changed$/);

    const mongoEntry = await getCollection('source_test1').findOne();
    expect(clean(mongoEntry)).toMatchSnapshot();
  });

  it('should not save if nothing changed', async () => {
    await mongoSetup.insertData({
      // eslint-disable-next-line camelcase
      source_test2: [
        {
          id: 'test2',
          commonID: 'test2',
          sequentialID: 0,
          data: {},
          date: new Date(0)
        }
      ]
    });

    await getEntryCopy('test2', {
      id: 'test2',
      commonID: 'test2',
      data: {},
      modificationDate: new Date(0)
    });

    const mongoEntry = await getCollection('source_test2').findOne();
    expect(clean(mongoEntry)).toMatchSnapshot();
  });
});
