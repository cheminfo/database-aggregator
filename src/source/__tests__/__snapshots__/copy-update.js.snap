// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source copy updates should not allow to change the commonID 1`] = `
Object {
  "commonID": "test1",
  "data": Object {},
  "date": 1970-01-01T00:00:00.000Z,
  "id": "test1",
  "sequentialID": 0,
}
`;

exports[`source copy updates should not save if nothing changed 1`] = `
Object {
  "commonID": "test2",
  "data": Object {},
  "date": 1970-01-01T00:00:00.000Z,
  "id": "test2",
  "sequentialID": 0,
}
`;
